### DISCUSSION TITLE
502 Bad Gateway

### DISCUSSION THREAD

**Discussion by clevercircuits** [requarks/wiki#6076] [Error / Bug Report]
Wiki.js is working fine under http://localhost:3000
Once Systemd Service installed
SSL certification created
Nginx installed and configured
SE Linux configured
Now the wiki.js stopped working on both ports 80 and 433

Error: "502 Bad Gateway" from Nginx

**Comment by fnafox** [comment]
If it works fine until those things are _configured_ that would indicate there's an error in your configuration, can't help without more info but my guess is a broken nginx reverse proxy configuration.

**Comment by NGPixel** [comment]
502 is a generic nginx reverse proxy error. Check your nginx configuration.

**Comment by clevercircuits** [comment]
I got further into the issue:
Nginx configuration is fine and Nginx works
wikijs service runs by starting it manually. But it crashes each time Linux (Rocky linux) reboots
Status of wikijs.service:
wikijs.service: Scheduled restart job, restart counter is at 5.
Jan 26 20:40:47 wikijs.localhost systemd[1]: Stopped Wiki.js.
Jan 26 20:40:47 wikijs.localhost  systemd[1]: wikijs.service: Start request repeated too quickly.
Jan 26 20:40:47 wikijs.localhost  systemd[1]: wikijs.service: Failed with result 'exit-code'.
Jan 26 20:40:47 wikijs.localhost  systemd[1]: Failed to start Wiki.js.
 
Any ideas?

**Reply by fnafox** [reply]
Does your service file have load order defined in the unit configuration?
```
[Unit]
After=network.target
```

**Reply by clevercircuits** [reply]
**This is my service file content:**
```
[Unit]
Description=Wiki.js
After=network.target

[Service]
Type=simple
ExecStart=/usr/bin/node server
Restart=always

User=ccwikiuser
Environment=NODE_ENV=production
WorkingDirectory=/var/www/wikijs

[Install]
WantedBy=multi-user.target
```

**Reply by fnafox** [reply]
Does the user ccwikiuser have permissions to the wiki directory?
Is your wiki -really- extracted into `/var/www/wikijs`? Working directory should be the one where the application was extracted. Normally you wouldn't place that in your public html directories.
What error does systemd give you (`journalctl -xeu wiki.service`)? Can you even start it with systemd? (`systemctl start wiki.service`)

**Reply by clevercircuits** [reply]
Okay
the ccwikiuser was set for the postgresql database.
I changed the User=root; because the /var/www/wikijs folder is owned by the root user.
Now the wikijs service works fine under http://localhost:3000
Nginx is running (Status = active)
journalctl -xeu wikijs.service delivers: A start job for unit wikijs.service has finished successfully)

HTTP requests with port 3000 are fine
HTTPS requests still end with "502 Bad Gateway"

Nginx configuration is correct

thanks
